### DISCUSSION TITLE
Some clarification needed regarding the nodeset entry

### DISCUSSION THREAD

**Discussion by RakshanPremsagar** [eclipse-milo/milo#888] [General]
Hello, I have a sample address space XML file having the entry as below 

```
<?xml version="1.0" encoding="UTF-8"?>
<UANodeSet xmlns="http://opcfoundation.org/UA/2011/03/UANodeSet.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:uax="http://opcfoundation.org/UA/2008/02/Types.xsd">
   <NamespaceUris>
      <Uri>http://uri1.org</Uri>
      <Uri>http://uri2.org</Uri>
      <Uri>http://uri3.org</Uri>
      <Uri>http://uri4.org</Uri>
      <Uri>http://uri5.org</Uri>
      <Uri>http://uri6.org</Uri>
      <Uri>http://uri7.org</Uri>
   </NamespaceUris>

<UAObject NodeId="ns=3;s=Boiler.Unit.Beh" BrowseName="7:Beh">           ------------------------(1)
      <DisplayName>Beh</DisplayName>
      <References>
</References>

<UAObject NodeId="ns=3;s=Boiler.Unit.Temp" BrowseName="3:Temp"> ------------------------(2)
      <DisplayName>Temp</DisplayName>
      <References>
</References>
   </UAObject>

```

My question here is,
What does BrowseName indicate here in (1)? I can understand that we have  a node present in ns 3, but does the browseName 7:Beh mean that the node Beh is an instance of the node present in ns=7?

Also in (2), does it mean that the Temp is a new object which is only defined in ns =3?

Please let me know.

Regards,
Rakshan

**Comment by arnoweiss** [comment]
Hi Rakshan,

concerning (1): I personally haven't seen that divergernce between ns from BrowseName and NodeId before. I think the node is modelled incorrectly since a node can only be present in one ns. Usually that would be the maximum NamespaceId (ns=7 in this case) so the NodeId should be "ns=7;s=Boiler.Unit.Beh". 

concerning (2): Yes, this describes a node in ns=3. However, a single Nodeset file should only define a single namespace and reference nodes from other namespaces via the NamespaceUris/RequiredModels lists. It's also strange to me that a Temperature would be modelled as a UAObject, it should probably be a UAVariable.

Let me guess: This is handwritten XML, right?

**Reply by RakshanPremsagar** [reply]
yes it is a handwritten one.. But i am having cases where I have observation similar to (1)..

**Comment by kevinherron** [comment]
I think it is okay that the BrowseName and NodeId are in separate namespaces.

Imagine one namespace that defines a bunch of standard type definitions and names for those types, e.g. a Boiler.

Then you have another namespace where you've added instances of Nodes that use those type definitions. It would not be strange in that case for the NodeId of the instance to be in a different namespace than the BrowseName, which would be from the namespace with all the type definitions.

**Reply by arnoweiss** [reply]
You're right. See [BrowseName](https://reference.opcfoundation.org/Core/docs/Part3/5.2.4/).
> Servers may often choose to use the same namespace for the NodeId and the BrowseName. However, if they want to provide a standard Property, its BrowseName shall have the namespace of the standards body although the namespace of the NodeId reflects something else, for example the local Server.
